src/main/main.ts

```diff
@@ -188,214 +189,195 @@ function initializeApp() {
-          console.log("Removed all temporary files from", tempDir);
-        } catch (e) {
-          console.warn(
-            "Failed to remove temporary files from",
-            tempDir,
-            "\n",
-            e
-          );
-        }
-        win?.webContents.send("update copying complete");
-      } else if (process.platform == "darwin") {
-        // .tar.{gz,bz2} 해제
-        const lowerFname = dlFname.toLowerCase();
-        const bz2 =
-          lowerFname.endsWith(".tar.bz2") || lowerFname.endsWith(".tbz");
-        console.log(
-          "Start to extract the tarball archive",
-          dlPath,
-          "to",
-          extractPath
-        );
-        try {
-          await spawnPromise(
-            "tar",
-            [`xvf${bz2 ? "j" : "z"}`, dlPath, "-C", extractPath],
-            { capture: ["stdout", "stderr"] }
-          );
-        } catch (e) {
-          console.error(`${e}:\n`, e.stderr);
-          throw e;
-        }
-        console.log(
-          "The tarball archive",
-          dlPath,
-          "has extracted to ",
-          extractPath
-        );
-      } else {
-        console.warn("Not supported platform.");
-        return;
-      }
+  if (win == null) return;
 
-      // 압축을 푼 뒤 압축 파일은 제거합니다.
-      await fs.promises.unlink(dlPath);
+  console.log("peerVersionExtra (hex):", peerVersionExtra);
+  const buffer = Buffer.from(peerVersionExtra, "hex");
+  console.log("peerVersionExtra (bytes):", buffer);
+  const extra = decode(buffer) as BencodexDict;
+  console.log("peerVersionExtra (decoded):", JSON.stringify(extra)); // 다른 프로세스라 잘 안보여서 JSON으로...
+  const macOSBinaryUrl = extra.get("macOSBinaryUrl") as string;
+  const windowsBinaryUrl = extra.get("WindowsBinaryUrl") as string;
+  console.log("macOSBinaryUrl: ", macOSBinaryUrl);
+  console.log("WindowsBinaryUrl: ", windowsBinaryUrl);
+  const downloadUrl =
+    process.platform === "win32"
+      ? windowsBinaryUrl
+      : process.platform === "darwin"
+      ? macOSBinaryUrl
+      : null;
 
-      // 설정 합치기
-      const newConfig = JSON.parse(
-        await fs.promises.readFile(configPath, { encoding: "utf-8" })
-      );
-      const config = {
-        ...bakConfig,
-        ...newConfig,
-      };
-      await fs.promises.writeFile(configPath, JSON.stringify(config), "utf-8");
+  if (downloadUrl == null) return; // 지원 안 하는 플랫폼이니 무시.
+
+  // TODO: 이어받기 되면 좋을 듯
+  const options: ElectronDLOptions = {
+    onStarted: (downloadItem: DownloadItem) => {
+      console.log("Starts to download:", downloadItem);
+    },
+    onProgress: (status: IDownloadProgress) => {
+      const percent = (status.percent * 100) | 0;
       console.log(
-        "The existing and new configuration files has been merged:",
-        config
+        `Downloading ${downloadUrl}: ${status.transferredBytes}/${status.totalBytes} (${percent}%)`
       );
+      win?.webContents.send("update download progress", status);
```

Comment:
No need to check null.
```suggestion
      win.webContents.send("update download progress", status);
```

Reply:
I guess it's still needed because TypeScript Compiler doesn't seem to know that it can't be null.

```
PS C:\Users\Swen Mun\Documents\electron-launcher> tsc -noEmit
src/main/main.ts:260:7 - error TS2531: Object is possibly 'null'.

260       win.webContents.send("update download progress", status);
          ~~~


Found 1 error
```
Of course, it can be solved by changing the type of `win`, but that seems to be a problem beyond the scope of this PR.

Reply:
Oh, yes. This scope is not the same context as outside the callback. But please consider other `win? -> win` suggestions in same context with `if (win === null) {}`.